nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -304,6 +304,113 @@ in {
           phone-numbers.
         '';
       };
+
+      objectstore = let
+        s3Arguments = {
+          bucket = mkOption {
+            type = types.str;
+            example = "nextcloud";
+            description = ''
+              The name of the S3 bucket.
+            '';
+          };
+          autocreate = mkOption {
+            type = types.bool;
+            description = ''
+              Create the objectstore if it does not exist.
+            '';
+          };
+          key = mkOption {
+            type = types.str;
+            example = "EJ39ITYZEUH5BGWDRUFY";
+            description = ''
+              The access key for the S3 bucket.
+            '';
+          };
+          secretFile = mkOption {
+            type = types.str;
+            example = "/var/nextcloud-objectstore-s3-secret";
+            description = ''
+              The full path to a file that contains the access secret. Must be
+              readable by user <literal>nextcloud</literal>.
+            '';
+          };
+          hostname = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            example = "example.com";
+            description = ''
+              Required for some non-Amazon implementations.
```

Comment:
Does that mean, it's for services that are not from Amazon, but speak the "S3 protocol"? Anyways, please give some more context here.

Reply:
I'm unsure about this! My limited understanding is that in some implementations of "S3-compatible" storage a hostname is required, whereas apparently it is not on Amazon? I'd have no idea having avoided Amazon, but I know it is required for my digital ocean space at least.

Reply:
I should add that my comment here provides just about as much clarity as [the nextcloud docs](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/primary_storage.html#simple-storage-service-s3) do.

Reply:
Yeah, makes sense.